src/components/CollectionsAndPalettes/common/UploadCard/UploadCard.tsx

```diff
@@ -1,14 +1,35 @@
 import * as React from "react";
 import styles from "components/CollectionsAndPalettes/common/UploadCard/UploadCard.module.scss";
 import cn from "classnames";
+import {
+  AppContext,
+  AppContextState,
+  Modals,
+} from "../../../../context/AppContext";
+
+interface Props {
+  caption: string;
+  hasIcon?: boolean;
+}
+
+export default class UploadCard extends React.Component<Props, any> {
+  static contextType = AppContext;
+  context!: AppContextState;
+  modalTarget = null;
+  createCollection = () => {
+    this.context.openModal(Modals.CreateCollection);
+  };
 
-export default class UploadCard extends React.Component<any, any> {
   render() {
     return (
       <div className={cn(styles.uploadContainer, "masonry-item")}>
-        <div>
-          <span>Upload a photo or drag & drop here </span>
-        </div>
+        {this.props.hasIcon && (
```

Comment:
We should make the entire card clickable, not just the icon, and also, in order to make it more reusable, add an onClick prop to it and let the parent handle opening the modal. That way, we can reuse the card in more places and we won't have to add logic inside the component for any cases where we might need it.